Bottom: a090421b8d9ae408d97fca61f6e230e5054a2e2d
Top:    bd664be5fecd03a44f074f035df966ccada05304
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-03-09 20:58:16 +1300

Refresh of stream-tagger

---

diff --git a/testtools/testresult/real.py b/testtools/testresult/real.py
index a7a0492..381f849 100644
--- a/testtools/testresult/real.py
+++ b/testtools/testresult/real.py
@@ -434,7 +434,7 @@ class StreamTagger(CopyStreamResult):
         self.discard = frozenset(discard or ())
 
     def status(self, *args, **kwargs):
-        test_tags = kwargs.get('test_tags', set())
+        test_tags = kwargs.get('test_tags') or set()
         test_tags.update(self.add)
         test_tags.difference_update(self.discard)
         kwargs['test_tags'] = test_tags or None
diff --git a/testtools/tests/test_testresult.py b/testtools/tests/test_testresult.py
index 7ffbcac..e99ae8d 100644
--- a/testtools/tests/test_testresult.py
+++ b/testtools/tests/test_testresult.py
@@ -664,11 +664,13 @@ class TestStreamTagger(TestCase):
         result.startTestRun()
         result.status()
         result.status(test_tags=set(['bar']))
+        result.status(test_tags=None)
         result.stopTestRun()
         self.assertEqual([
             ('startTestRun',),
             ('status', None, None, set(['foo']), True, None, None, False, None, None, None),
             ('status', None, None, set(['foo', 'bar']), True, None, None, False, None, None, None),
+            ('status', None, None, set(['foo']), True, None, None, False, None, None, None),
             ('stopTestRun',),
             ], log._events)
 
@@ -677,6 +679,7 @@ class TestStreamTagger(TestCase):
         result = StreamTagger([log], discard=['foo'])
         result.startTestRun()
         result.status()
+        result.status(test_tags=None)
         result.status(test_tags=set(['foo']))
         result.status(test_tags=set(['bar']))
         result.status(test_tags=set(['foo', 'bar']))
@@ -685,6 +688,7 @@ class TestStreamTagger(TestCase):
             ('startTestRun',),
             ('status', None, None, None, True, None, None, False, None, None, None),
             ('status', None, None, None, True, None, None, False, None, None, None),
+            ('status', None, None, None, True, None, None, False, None, None, None),
             ('status', None, None, set(['bar']), True, None, None, False, None, None, None),
             ('status', None, None, set(['bar']), True, None, None, False, None, None, None),
             ('stopTestRun',),
